### DISCUSSION TITLE
Is there a way to check that all schemas have descriptions and examples?

### DISCUSSION THREAD

**Discussion by rafasf** [stoplightio/spectral#1487] [Q&A]
I'm having a hard time on creating a rule that checks if the schemas properties have description and example. The more important being the description.

Does anyone know if that is a thing I could really do?

I only explored the built-in rules so far, but no success.

**Comment by gfiehler** [comment]
I have not tried this but I have a similar rule so I think this would work.

"all-schemas-have-desc": {
      "description": "All schema have descriptions.",
      "message": "All schema must have a description.",
      "recommended": true,
      "type": "validation",
      "resolved": false,
      "given": "$.components.schemas..description",
      "severity": "warn",
      "then": {
          "function": "truthy"
        }
    },
or you can add a pattern and test with regex that some text exists.
